### PR TITLE
Update minSdk in AndroidManifest

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Update minSdk in `AndroidManifest.xml` from some modules.

#### WHY

In order to match changes done in https://github.com/google/horologist/pull/1082 and https://github.com/google/horologist/pull/1083.

#### Checklist :clipboard:
- [N/A] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [N/A] Update metalava's signature text files

### CODE REVIEW

File: datalayer-phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt [hunk 1]
```diff
@@ -57,7 +57,7 @@ public class PhoneDataLayerAppHelper(context: Context) :
          * example, Samsung devices report the plugin packages that handle comms with GW4, GW5
          * etc, whereas the package name for the companion *app* is different.
          */
-        val launchPackage = companionLookup.getOrDefault(companionPackage, companionPackage)
```
Comment (luizgrp) [inline datalayer-phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt:60]: `getOrDefault` requires API level 24

File: datalayer/src/main/java/com/google/android/horologist/data/DataLayerAppHelper.kt [hunk 2]
```diff
@@ -144,14 +147,22 @@ abstract class DataLayerAppHelper(protected val context: Context) {
...
+        surfacePrefix: String,
+        capabilities: Map<String, CapabilityInfo>
+    ): Map<String, Set<String>> {
         val idToSurfaceSet = mutableMapOf<String, Set<String>>()
         capabilities
             .entries.filter { it.key.startsWith(surfacePrefix) }
             .forEach { entry ->
                 val name = entry.key.removePrefix(surfacePrefix)
                 for (node in entry.value.nodes) {
-                    idToSurfaceSet.merge(node.id, setOf(name)) { s1, s2 -> s1 + s2 }
```
Comment (luizgrp) [inline datalayer/src/main/java/com/google/android/horologist/data/DataLayerAppHelper.kt:154]: `merge` requires API level 24

File: datalayer-phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt (hunk 1 above)
Comment (yschimke) [reply datalayer-phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt:60]: or I guess change the desugaring? https://developer.android.com/studio/write/java8-support-table

**Review by yschimke — APPROVED** [review]

File: datalayer-phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt (hunk 1 above)
Comment (luizgrp) [reply datalayer-phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt]: tried in 0fddaf03a22623d785e479b86f178e0846e660ad but still [failed](https://github.com/google/horologist/actions/runs/4414494565/jobs/7736281159):

```
/home/runner/work/horologist/horologist/datalayer-phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt:60: Error: Call requires API level 24 (current min is 23): java.util.Map#getOrDefault [NewApi]
        val launchPackage = companionLookup.getOrDefault(companionPackage, companionPackage)
``` 

raised https://github.com/google/horologist/issues/1100

**Pull request merged**